Tests/GRPCHTTP2CoreTests/Client/Connection/Connection+Equatable.swift

```diff
@@ -48,15 +47,45 @@ extension Connection.CloseReason: Equatable {
       (.remote, .remote):
       return true
 
-    case (.error(let lhsError), .error(let rhsError)):
+    case (.error(let lhsError, let lhsStreams), .error(let rhsError, let rhsStreams)):
       if let lhs = lhsError as? RPCError, let rhs = rhsError as? RPCError {
-        return lhs == rhs
+        return lhs == rhs && lhsStreams == rhsStreams
       } else {
-        return true
+        return lhsStreams == rhsStreams
       }
 
     default:
       return false
     }
   }
 }
+
+extension ClientConnectionEvent: Equatable {
+  public static func == (lhs: ClientConnectionEvent, rhs: ClientConnectionEvent) -> Bool {
+    switch (lhs, rhs) {
+    case (.ready, .ready):
+      return true
+    case (.closing(let lhsReason), .closing(let rhsReason)):
+      return lhsReason == rhsReason
+    default:
+      return false
+    }
+  }
+}
+
+extension ClientConnectionEvent.CloseReason: Equatable {
+  public static func == (lhs: Self, rhs: Self) -> Bool {
+    switch (lhs, rhs) {
+    case (.goAway(let lhsCode, let lhsMessage), .goAway(let rhsCode, let rhsMessage)):
+      return lhsCode == rhsCode && lhsMessage == rhsMessage
+    case (.unexpected(_, let lhsIsIdle), .unexpected(_, let rhsIsIdle)):
+      return lhsIsIdle == rhsIsIdle
```

Comment:
Do we want to do the same as above for `Connection.CloseReason` and compare the errors if they're `RPCError`s (or `Equatable`)?

Reply:
We can't compare them as `Equatable`s, but we should do the `RPCError` comparison as above. I'll do that.